src/middlewared/middlewared/plugins/crypto.py

```diff
@@ -741,7 +741,7 @@ def __init__(self, *args, **kwargs):
             ))
 
         if not os.path.exists(root_path):
-            os.mkdir(root_path, 0o755)
+            os.makedirs(root_path, 0o755, exist_ok=True)
```

Comment:
`if not os.path.exists(root_path)` check has no sense anymore

Reply:
Not true, there could be a race condition.

Reply:
Agreed :)

Reply:
What race condition? `exist_ok=True`

Reply:
Yes, exist_ok=True handles the race condition and its the only reason its being used.

I prefer to check for the existence before blindly calling makedirs and raising an exception. That does not make os.path.exists useless, it avoids extra calls/checks and an exception handling.